cmake/OpenCVFindLibsGrfmt.cmake

```diff
@@ -34,7 +34,7 @@ if(NOT ZLIB_FOUND)
   set(ZLIB_INCLUDE_DIRS ${ZLIB_INCLUDE_DIR})
   set(ZLIB_LIBRARIES ${ZLIB_LIBRARY})
 
-  ocv_parse_header2(ZLIB "${${ZLIB_LIBRARY}_SOURCE_DIR}/zlib.h" ZLIB_VERSION)
+  ocv_parse_header_version(ZLIB "${${ZLIB_LIBRARY}_SOURCE_DIR}/zlib.h" ZLIB_VERSION)
```

Comment:
macro has 2 parameters.

Where is specified `ZLIB_VERSION`?

Reply:
Fixed

Reply:
We still missing `ZLIB_VERSION`.

CMake is switching to `_VERSION` variables. Others are mostly deprecated.
Perhaps we should do the same.

```
$ grep -R ZLIB_VERSION /usr/share/cmake/
/usr/share/cmake/Modules/FindZLIB.cmake:``ZLIB_VERSION``
/usr/share/cmake/Modules/FindZLIB.cmake:  See also legacy variable ``ZLIB_VERSION_STRING``.
/usr/share/cmake/Modules/FindZLIB.cmake:``ZLIB_VERSION_MAJOR``
/usr/share/cmake/Modules/FindZLIB.cmake:    Superseded by ``ZLIB_VERSION``.
/usr/share/cmake/Modules/FindZLIB.cmake:``ZLIB_VERSION_MINOR``
/usr/share/cmake/Modules/FindZLIB.cmake:    Superseded by ``ZLIB_VERSION``.
/usr/share/cmake/Modules/FindZLIB.cmake:``ZLIB_VERSION_PATCH``
/usr/share/cmake/Modules/FindZLIB.cmake:    Superseded by ``ZLIB_VERSION``.
/usr/share/cmake/Modules/FindZLIB.cmake:``ZLIB_VERSION_TWEAK``
/usr/share/cmake/Modules/FindZLIB.cmake:    Superseded by ``ZLIB_VERSION``.
/usr/share/cmake/Modules/FindZLIB.cmake:``ZLIB_VERSION_STRING``
/usr/share/cmake/Modules/FindZLIB.cmake:    Superseded by ``ZLIB_VERSION``.
/usr/share/cmake/Modules/FindZLIB.cmake:  The major version of zlib.  Superseded by ``ZLIB_VERSION_MAJOR``.
/usr/share/cmake/Modules/FindZLIB.cmake:  The minor version of zlib.  Superseded by ``ZLIB_VERSION_MINOR``.
/usr/share/cmake/Modules/FindZLIB.cmake:  The patch version of zlib.  Superseded by ``ZLIB_VERSION_PATCH``.
/usr/share/cmake/Modules/FindZLIB.cmake:  file(STRINGS "${ZLIB_INCLUDE_DIR}/zlib.h" ZLIB_H REGEX "^#define ZLIB_VERSION \"[^\"]*\"$")
/usr/share/cmake/Modules/FindZLIB.cmake:  if(ZLIB_H MATCHES "ZLIB_VERSION \"(([0-9]+)\\.([0-9]+)(\\.([0-9]+)(\\.([0-9]+))?)?)")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_STRING "${CMAKE_MATCH_1}")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_MAJOR "${CMAKE_MATCH_2}")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_MINOR "${CMAKE_MATCH_3}")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_PATCH "${CMAKE_MATCH_5}")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_TWEAK "${CMAKE_MATCH_7}")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_STRING "")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_MAJOR "")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_MINOR "")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_PATCH "")
/usr/share/cmake/Modules/FindZLIB.cmake:    set(ZLIB_VERSION_TWEAK "")
/usr/share/cmake/Modules/FindZLIB.cmake:  set(ZLIB_MAJOR_VERSION "${ZLIB_VERSION_MAJOR}")
/usr/share/cmake/Modules/FindZLIB.cmake:  set(ZLIB_MINOR_VERSION "${ZLIB_VERSION_MINOR}")
/usr/share/cmake/Modules/FindZLIB.cmake:  set(ZLIB_PATCH_VERSION "${ZLIB_VERSION_PATCH}")
/usr/share/cmake/Modules/FindZLIB.cmake:  set(ZLIB_VERSION "${ZLIB_VERSION_STRING}")
/usr/share/cmake/Modules/FindZLIB.cmake:                                       VERSION_VAR ZLIB_VERSION
```

Reply:
Set both `ZLIB_VERSION` and `ZLIB_VERSION_STRING`? `ZLIB_VERSION` is introduced in cmake 3.26 which is quite modern. I'm not sure how much compatibility we should keep. Should we set other variables like `ZLIB_MAJOR_VERSION`? CMake(master branch)'s FindPNG module doesn't use these version string.